### PR TITLE
Hosting startup topic+sample update

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #5495 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/host/platform-specific-configuration?view=aspnetcore-2.1&branch=pr-en-us-11374)

* Topic to 2.1+
* Samples to 2.2

### CODE REVIEW

File: aspnetcore/fundamentals/host/platform-specific-configuration/samples/2.x/HostingStartupApp/Pages/Error.cshtml [hunk 1]
```diff
@@ -19,5 +19,8 @@
     Swapping to the <strong>Development</strong> environment displays detailed information about the error that occurred.
 </p>
 <p>
-    <strong>The Development environment shouldn't be enabled for deployed applications</strong>. It can result in displaying sensitive information from exceptions to end users. For local debugging, enable the Development environment by setting the <strong>ASPNETCORE_ENVIRONMENT</strong> environment variable to <strong>Development</strong> and restarting the app or adding <code>.UseEnvironment("Development")</code> to <code>WebHost</code> in <i>Program.cs</i> and restarting the app.
+    <strong>The Development environment shouldn't be enabled for deployed applications.</strong>
+    It can result in displaying sensitive information from exceptions to end users.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/host/platform-specific-configuration/samples/2.x/HostingStartupApp/Pages/Error.cshtml]: Using the Development environment in production can leak sensitive information, for example from exceptions.

Comment (guardrex) [reply aspnetcore/fundamentals/host/platform-specific-configuration/samples/2.x/HostingStartupApp/Pages/Error.cshtml]: That's part of the template :point_right: https://github.com/aspnet/AspNetCore/blob/master/src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Error.cshtml#L23

Comment (guardrex) [reply aspnetcore/fundamentals/host/platform-specific-configuration/samples/2.x/HostingStartupApp/Pages/Error.cshtml]: There are 34 additional cases of that line across our topic samples, do you want me to proceed with the change here? If so, I'll patch all 35 instances on this PR.

**Comment by Rick-Anderson** [comment]
Fix this with an include and S&M. Create an issue for the other ones and fix when you have time.

**Comment by Rick-Anderson** [comment]
If that's in the template. leave it. I'll open a template bug.

**Comment by guardrex** [comment]
Yes, it's in the template for the *Error.cshtml* pages of the templates. This will give you the links to their templates that have it ...

https://github.com/aspnet/AspNetCore/search?q=%22It+can+result+in+displaying%22&unscoped_q=%22It+can+result+in+displaying%22

**Pull request merged**